azure-functions-powershell-library/BundledBindings/IOutputBinding.cs

```diff
@@ -0,0 +1,20 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation.Language;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AzureFunctions.PowerShell.SDK.BundledBindings
+{
+    public abstract class IOutputBinding : IBinding
+    {
+        public const BindingInformation.Directions BindingDirection = BindingInformation.Directions.Out;
+        public abstract BindingInformation ExtractBinding(AttributeAst attribute);
```

Comment:
Since `ExtractBinding` seems to be used in both input and output bindings, does it make sense to just declare it at the `IBinding` level?

Reply:
Signatures are different - no ParameterAst is available for output bindings, since they are attributes on the param block, not individual parameters. There might be value in changing the signature to (AttributeAst attribute, ParameterAst? parameter) and moving it to IBinding, to make this a soft requirement, but then each binding will have to do null checks on the parameter before using it

Reply:
If the signatures are different, which I had not noticed, then I'm ok to leave as-is :-)